x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/sourceonly/SourceOnlySnapshotRepository.java

```diff
@@ -148,7 +148,15 @@ public void snapshotShard(SnapshotShardContext context) {
         final Store store = context.store();
         Directory unwrap = FilterDirectory.unwrap(store.directory());
         if (unwrap instanceof FSDirectory == false) {
-            throw new AssertionError("expected FSDirectory but got " + unwrap.toString());
+            context.onFailure(
+                new IllegalStateException(
+                    context.indexCommit()
+                        + " is not a regular index and cannot be snapshotted into a source-only repository ["
+                        + unwrap.toString()
+                        + ']'
```

Comment:
I believe `unwrap` directory is related to the index type, not a repository, is it? If so may be it is better to rephrase this to:
```
context.indexCommit() + " is not a regular index, but ["+ unwrap.toString()+"] and cannot be snapshotted into a source-only repository"
```

Reply:
Thanks for the suggestion, I'll borrow it.